hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementHelper.java

```diff
@@ -25,11 +25,18 @@
  * @author Steve Ebersole
  */
 public class EnhancementHelper {
+
+	@FunctionalInterface
+	public interface InheritanceChecker {
+		boolean hasSubclasses(String entityName);
+	}
+
 	/**
 	 * Should the given property be included in the owner's base fetch group?
 	 */
 	public static boolean includeInBaseFetchGroup(
 			Property bootMapping,
+			InheritanceChecker inheritanceChecker,
 			boolean isEnhanced,
 			boolean allowEnhancementAsProxy,
 			boolean collectionsInDefaultFetchGroupEnabled) {
```

Comment:
I'm assuming that changing this SPI is OK in master.

Note to self: when this gets backported, the old version of the method should be deprecated.

Reply:
I really cannot imagine user call this.  But sure, whatever you think wrt 5.3